application/clicommands/ScanCommand.php

```diff
@@ -58,9 +66,66 @@ public function indexAction()
                 $name
             );
 
+            if ($sinceLastScan !== null) {
+                $this->cleanupLastScanTargets($this->getDb(), $sinceLastScan);
+            }
+
             $verified = CertificateUtils::verifyCertificates($this->getDb());
 
+            if ($sinceLastSeen !== null) {
+                $this->cleanupLastSeenTargets($this->getDb(), $sinceLastSeen);
+                CertificateUtils::cleanupCertificates($this->getDb(), $sinceLastSeen);
+            }
+
+            CertificateUtils::cleanupCertificateChains($this->getDb());
             Logger::info("Checked %d certificate chain%s.", $verified, $verified !== 1 ? 's' : '');
         }
     }
+
+    protected function cleanupLastScanTargets(Connection $db, DateTime $sincLastScan)
+    {
+        $db->delete(
+            'x509_target',
+            ['last_scan < ?' => $sincLastScan->getTimestamp()]
+        );
+    }
+
+    protected function cleanupLastSeenTargets(Connection $db, DateTime $sincLastSeen)
+    {
+        $db->delete(
+            'x509_target',
+            [new Expression(sprintf(
+                "last_seen < last_scan - %d",
+                (new DateTime())->getTimestamp() - $sincLastSeen->getTimestamp()
+            ))]
+        );
+    }
+
+    /**
+     * @param string|null $threshold
+     * @return DateTime|null
+     */
+    protected function processThresholdToDateTiime(?string $threshold)
```

Comment:
Can you please put this in a trait? This will also be needed by some other cli commands.